From 0e2817dc66c4f8f099383d6dbf6bdb4cc18a93e5 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:23:51 -0400
Subject: [tools] perf: Clean up util/include/linux/compiler.h

Message-id: <1408462094-14194-63-git-send-email-jolsa@redhat.com>
Patchwork-id: 88069
O-Subject: [PATCH RHEL7.1 BZ1131394 062/325] tools/perf: Clean up util/include/linux/compiler.h
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 7a10822a30060eceddda16c051086c00acb449c9
Author: Ingo Molnar <mingo@kernel.org>
Date: Mon Oct 7 11:26:18 2013 +0200

description
===========
Use the standard CPP style we use in the kernel:

 #ifndef foo
 # define foo bar
 #endif
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/include/linux/compiler.h b/tools/perf/util/include/linux/compiler.h
index 96b919d..bef4d3d 100644
--- a/tools/perf/util/include/linux/compiler.h
+++ b/tools/perf/util/include/linux/compiler.h
@@ -2,20 +2,25 @@
 #define _PERF_LINUX_COMPILER_H_
 
 #ifndef __always_inline
-#define __always_inline inline
+# define __always_inline inline __attribute__((always_inline))
 #endif
+
 #define __user
+
 #ifndef __attribute_const__
-#define __attribute_const__
+# define __attribute_const__
 #endif
 
 #ifndef __maybe_unused
-#define __maybe_unused  __attribute__((unused))
+# define __maybe_unused  __attribute__((unused))
+#endif
+
+#ifndef __packed
+# define __packed  __attribute__((__packed__))
 #endif
-#define __packed __attribute__((__packed__))
 
 #ifndef __force
-#define __force
+# define __force
 #endif
 
 #endif
-- 
1.7.1